overlord/state/state.go

```diff
@@ -110,18 +114,21 @@ type State struct {
 
 // New returns a new empty state.
 func New(backend Backend) *State {
```

Comment:
should `ReadState` also be updated to initialize the condition and map you've added?

Reply:
I think the map is initialized by the unmarshal code, but you are right the code needs to take care of the condition? I wonder whether pebble has the same bug?

Reply:
anyway we need some code that tries to do WaitNotices on a ReadState result

Reply:
Really good catch, thanks! The PR has been submitted and merged into pebble https://github.com/canonical/pebble/pull/320, I'll include the same changes here.

Reply:
Thank you for taking care of that @olivercalder